OptimizelySDK.Tests/OdpTests/GraphQLManagerTest.cs

```diff
@@ -0,0 +1,312 @@
+                               "\"state\":\"qualified\"}},{\"node\":{\"name\":" +
+                               "\"has_email_opted_in\",\"state\":\"qualified\"}}]}}}}";
+            _mockOdpClient.Setup(
+                    c => c.QuerySegments(It.IsAny<QuerySegmentsParameters>())).
+                Returns(responseData);
+            var manager = new GraphQLManager(_mockLogger.Object, _mockOdpClient.Object);
+
+            var segments = manager.FetchSegments(
+                VALID_ODP_PUBLIC_KEY,
+                ODP_GRAPHQL_URL,
+                FS_USER_ID,
+                "tester-101",
+                _segmentsToCheck);
+
+            Assert.IsTrue(segments.Length == 2);
+            Assert.Contains("has_email", segments);
+            Assert.Contains("has_email_opted_in", segments);
+            _mockLogger.Verify(l => l.Log(LogLevel.WARN, It.IsAny<string>()), Times.Never);
+        }
+
+        [Test]
+        public void ShouldHandleEmptyQualifiedSegments()
+        {
+            var responseData = "{\"data\":{\"customer\":{\"audiences\":" +
+                               "{\"edges\":[ ]}}}}";
+            _mockOdpClient.Setup(
+                    c => c.QuerySegments(It.IsAny<QuerySegmentsParameters>())).
+                Returns(responseData);
+            var manager = new GraphQLManager(_mockLogger.Object, _mockOdpClient.Object);
+
+            var segments = manager.FetchSegments(
+                VALID_ODP_PUBLIC_KEY,
+                ODP_GRAPHQL_URL,
+                FS_USER_ID,
+                "tester-101",
+                _segmentsToCheck);
+
+            Assert.IsTrue(segments.Length == 0);
+            _mockLogger.Verify(l => l.Log(LogLevel.WARN, It.IsAny<string>()), Times.Never);
+        }
+
+        [Test]
+        public void ShouldHandleErrorWithInvalidIdentifier()
+        {
+            var responseData = "{\"errors\":[{\"message\":" +
+                               "\"Exception while fetching data (/customer) : " +
+                               "java.lang.RuntimeException: could not resolve _fs_user_id = invalid-user\"," +
+                               "\"locations\":[{\"line\":1,\"column\":8}],\"path\":[\"customer\"]," +
+                               "\"extensions\":{\"classification\":\"DataFetchingException\"}}]," +
+                               "\"data\":{\"customer\":null}}";
+            _mockOdpClient.Setup(
+                    c => c.QuerySegments(It.IsAny<QuerySegmentsParameters>())).
+                Returns(responseData);
+            var manager = new GraphQLManager(_mockLogger.Object);
+
+            var segments = manager.FetchSegments(
+                VALID_ODP_PUBLIC_KEY,
+                ODP_GRAPHQL_URL,
+                FS_USER_ID,
+                "invalid-user", // invalid user
+                _segmentsToCheck);
+
+            Assert.IsTrue(segments.Length == 0);
+            _mockLogger.Verify(l => l.Log(LogLevel.WARN, It.IsAny<string>()), Times.Once);
```

Comment:
Can we check if all error log messages include "Audience segments fetch failed (...)"?